pallets/confidential/src/rng.rs

```diff
@@ -0,0 +1,118 @@
+use rand_core::{CryptoRng, Error, RngCore};
+use sp_runtime_interface::runtime_interface;
+
+#[cfg(not(feature = "std"))]
+use core::num::NonZeroU32;
+
+#[cfg(feature = "std")]
+use rand::rngs::OsRng;
+
+/// Host functions to allow native access to OS RNG from Wasm.
+///
+/// # Change Log
+/// - Runtime Spec version 1005. It is incompatible with previous versions, and it requires a
+/// binary update.
+///
+/// # TODO
+///  - Use lazy static RNG object to mitigate performance impact during the `OsRng` object
+///  creation. We should double-check the security implications of having a global RNG in a
+///  multi-threading environment.
+#[runtime_interface]
+pub trait NativeRng {
+    fn os_next_u32() -> u32 {
+        OsRng::default().next_u32()
+    }
+
+    fn os_next_u64() -> u64 {
+        OsRng::default().next_u64()
+    }
+
+    fn fill_bytes(dest: &mut [u8]) {
+        OsRng::default().fill_bytes(dest)
+    }
+
+    fn try_fill_bytes(dest: &mut [u8]) -> i32 {
+        match OsRng::default().try_fill_bytes(dest) {
+            Ok(..) => 0,
+            Err(err) => err.raw_os_error().unwrap_or(1),
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct Rng {
+    #[cfg(feature = "std")]
+    inner: OsRng,
+}
+
+impl Default for Rng {
+    #[cfg(feature = "std")]
+    fn default() -> Self {
+        Rng {
+            inner: OsRng::default(),
+        }
+    }
+
+    #[cfg(not(feature = "std"))]
+    fn default() -> Self {
+        Rng {}
```

Comment:
Can you still create `OsRng::default()` only once per extrinsic? Otherwise you are creating it multiple times if you need randomness multiple times.

Reply:
It is not as easy as we think at first sight. That is related with the **TODO** section, but the problem is that this structure has to be also represented in WASM, where `OsRng` does not exist. So we cannot store it as a member of this `wrapper`.